### PR TITLE
docs: add websocket rate limit example

### LINKED ISSUE #7521: docs: use the limit-conn plugin to limit the rate of websocket
### Current State

Users can use APISIX's limit-conn plugin to limit the rate of new websocket requests, but there is no relevant documentation to describe it.
A typical application scenario is a chatbot that communicates with a WebSocket server. In order not to be overwhelmed by requests from bots, the rate of their requests needs to be limited.

### Desired State

Add some examples or best practices of using the limit-conn plugin to limit the rate of websockets in the documentation.

### PR DESCRIPTION
**Pull request by soulbird**
### Description

Fixes #7521 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/plugins/limit-conn.md [hunk 1]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
     }
 }'
 ```
+
+## Limit the rate of WebSocket connections
+
+Apache APISIX supports WebSocket proxy, we can use limit-conn plugin to limit the number of concurrent WebSocket connections.
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/limit-conn.md:147]:
```suggestion
Apache APISIX supports WebSocket proxy, we can use `limit-conn` plugin to limit the number of concurrent WebSocket connections.
```

File: docs/en/latest/plugins/limit-conn.md [hunk 2]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+
+1. Start WebSocket Server
+
+````
+docker run -p 1980:8080 --name websocket-demo casperklein/websocket-demo
+````
+
+2. Register the route, enable the WebSocket proxy on the route and enable the limit-conn plugin
+
+````
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/limit-conn.md:157]:
```suggestion
````shell
```

File: docs/en/latest/plugins/limit-conn.md [hunk 3]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+        }
+    }
+}'
+````
+
+The above route enables the WebSocket proxy on `/ws`, and limits the number of concurrent WebSocket connections to 1. More than 1 concurrent WebSocket connection will return 503 to reject the request.
+
+3. Initiate a WebSocket request, and the link is established successfully
+
+````
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/limit-conn.md:185]:
```suggestion
````shell
```

File: docs/en/latest/plugins/limit-conn.md [hunk 4]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
 }'
 ```
+
+## Limit the rate of WebSocket connections
+
+Apache APISIX supports WebSocket proxy, we can use limit-conn plugin to limit the number of concurrent WebSocket connections.
+
+1. Start WebSocket Server
+
+````
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/limit-conn.md:151]:
```suggestion
````shell
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 5]
```diff
@@ -39,7 +39,7 @@ title: limit-conn
 | rejected_msg       | string | 否                                |            | 非空                                          | 当请求超过 `conn` + `burst` 这个阈值时，返回的响应体。                                                                                                                                                                     |
 | allow_degradation              | boolean  | 否                                | false       |                                                                     | 当插件功能临时不可用时是否允许请求继续。当值设置为 true 时则自动允许请求继续，默认值是 false。                                                                                                                                                    |
 
-### 如何启用
+## 开启插件
 
 下面是一个示例，在指定的 route 上开启了 limit-conn 插件，并设置 `key_type` 为 `var`:
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/limit-conn.md:44]:
```suggestion
下面是一个示例，在指定的路由上开启了 `limit-conn` 插件，并设置 `key_type` 为 `var`:
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 6]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
 ```
 
 现在就已经移除了 limit-conn 插件了。其他插件的开启和移除也是同样的方法。
+
+## 对 WebSocket 链接限速
+
+Apache APISIX 支持 WebSocket 代理，我们可以使用 limit-conn 插件限制 WebSocket 链接的并发数。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/limit-conn.md:142]:
```suggestion
Apache APISIX 支持 WebSocket 代理，我们可以使用 `limit-conn` 插件限制 WebSocket 链接的并发数。
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 7]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+
+1、启动 WebSocket Server
+
+```
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/limit-conn.md:146]:
```suggestion
```shell
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 8]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+docker run -p 1980:8080 --name websocket-demo casperklein/websocket-demo
+```
+
+2、注册路由，在路由上启用 WebSocket 代理并开启 limit-conn 插件
+
+```
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/limit-conn.md:152]:
```suggestion
```shell
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 9]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+        }
+    }
+}'
+```
+
+上述路由在 `/ws` 上开启了 WebSocket 代理，并限制了 WebSocket 链接并发数 1，超过 1 个并发的 WebSocket 链接将返回 503 拒绝请求。
+
+3、发起 WebSocket 请求，链接建立成功
+
+```
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/limit-conn.md:180]:
```suggestion
```shell
```

**Review by SylviaBABY — COMMENTED** [review]
I only review the contents you just adjusted, and the rest of the Chinese-related content adjustments will be carried out by the students of the OSPP project in the future.

File: docs/zh/latest/plugins/limit-conn.md [hunk 10]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
 ```
 
 现在就已经移除了 limit-conn 插件了。其他插件的开启和移除也是同样的方法。
+
+## 对 WebSocket 链接限速
```
Comment (juzhiyuan) [inline docs/zh/latest/plugins/limit-conn.md:140]:
```suggestion
## 对 WebSocket 连接限速
```

Comment (juzhiyuan) [reply docs/zh/latest/plugins/limit-conn.md:140]: Hey @navendu-pottekkat and @hf400159, please have a review when available 🙏

File: docs/zh/latest/plugins/limit-conn.md [hunk 11]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+
+Apache APISIX 支持 WebSocket 代理，我们可以使用 `limit-conn` 插件限制 WebSocket 连接的并发数。
+
+1、启动 WebSocket Server
+
+```shell
+docker run -p 1980:8080 --name websocket-demo casperklein/websocket-demo
+```
+
+2、注册路由，在路由上启用 WebSocket 代理并开启 limit-conn 插件
```
Comment (hf400159) [inline docs/zh/latest/plugins/limit-conn.md:150]:
```suggestion
2、创建路由并启用 WebSocket 代理以及配置 `limit-conn` 插件。
```

File: docs/en/latest/plugins/limit-conn.md [hunk 12]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+
+Apache APISIX supports WebSocket proxy, we can use `limit-conn` plugin to limit the number of concurrent WebSocket connections.
+
+1. Start WebSocket Server
+
+````shell
+docker run -p 1980:8080 --name websocket-demo casperklein/websocket-demo
+````
+
+2. Register the route, enable the WebSocket proxy on the route and enable the limit-conn plugin
```
Comment (hf400159) [inline docs/en/latest/plugins/limit-conn.md:155]:
```suggestion
2. Create a Route and enable the WebSocket proxy and the `limit-conn` plugin.
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 13]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+     --header "Sec-WebSocket-Version: 13" \
+     --http1.1 \
+     http://127.0.0.1:9080/ws
+
+HTTP/1.1 101 Switching Protocols
+Connection: upgrade
+Upgrade: websocket
+Sec-WebSocket-Accept: HSmrc0sMlYUkAGmm5OPpG2HaGWk=
+Server: APISIX/2.15.0
+```
```
Comment (hf400159) [inline docs/zh/latest/plugins/limit-conn.md:195]: **command** and **response** are represented by two code blocks.

File: docs/zh/latest/plugins/limit-conn.md [hunk 14]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+Server: APISIX/2.15.0
+
+<html>
+<head><title>503 Service Temporarily Unavailable</title></head>
+<body>
+<center><h1>503 Service Temporarily Unavailable</h1></center>
+<hr><center>openresty</center>
+</body>
+</html>
+```
```
Comment (hf400159) [inline docs/zh/latest/plugins/limit-conn.md:214]: ditto

File: docs/zh/latest/plugins/limit-conn.md [hunk 15]
```diff
@@ -136,3 +136,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    }
+}'
+```
+
+上述路由在 `/ws` 上开启了 WebSocket 代理，并限制了 WebSocket 连接并发数 1，超过 1 个并发的 WebSocket 连接将返回 503 拒绝请求。
```
Comment (hf400159) [inline docs/zh/latest/plugins/limit-conn.md:176]:
```suggestion
上述路由在 `/ws` 上开启了 WebSocket 代理，并限制了 WebSocket 连接并发数 1，超过 1 个并发的 WebSocket 连接将返回 `503` 拒绝请求。
```

File: docs/en/latest/plugins/limit-conn.md [hunk 16]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+     --header "Sec-WebSocket-Version: 13" \
+     --http1.1 \
+     http://127.0.0.1:9080/ws
+
+HTTP/1.1 101 Switching Protocols
+Connection: upgrade
+Upgrade: websocket
+Sec-WebSocket-Accept: HSmrc0sMlYUkAGmm5OPpG2HaGWk=
+Server: APISIX/2.15.0
+````
```
Comment (hf400159) [inline docs/en/latest/plugins/limit-conn.md:200]: **command** and **response** are represented by two code blocks.

File: docs/en/latest/plugins/limit-conn.md [hunk 17]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+Connection: keep-alive
+Server: APISIX/2.15.0
+
+<html>
+<head><title>503 Service Temporarily Unavailable</title></head>
+<body>
+<center><h1>503 Service Temporarily Unavailable</h1></center>
+<hr><center>openresty</center>
+</body>
+</html>
```
Comment (hf400159) [inline docs/en/latest/plugins/limit-conn.md:218]: ditto

File: docs/en/latest/plugins/limit-conn.md [hunk 18]
```diff
@@ -141,3 +141,79 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
     }
 }'
 ```
+
+## Limit the rate of WebSocket connections
```
Comment (hf400159) [inline docs/en/latest/plugins/limit-conn.md:145]: I think you can simplify this step. We don't have to tell the user how to create a WebSocket Server.

File: docs/zh/latest/plugins/limit-conn.md (hunk 14 above)
Comment (soulbird) [reply docs/zh/latest/plugins/limit-conn.md:214]: Both of these are response data

**Review by hf400159 — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Comment by moonming** [comment]
Do we have documentation for WebSocket? I feel that this PR should not be submitted to the limit conn part

File: docs/en/latest/plugins/limit-conn.md [hunk 19]
```diff
@@ -141,3 +141,76 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+
+Apache APISIX supports WebSocket proxy, we can use `limit-conn` plugin to limit the number of concurrent WebSocket connections.
+
+1. Create a Route and enable the WebSocket proxy and the `limit-conn` plugin.
```
Comment (SkyeYoung) [inline docs/en/latest/plugins/limit-conn.md:149]:
```suggestion
1. Create a Route, enable the WebSocket proxy and the `limit-conn` plugin.
```

too much `and`

**Comment by soulbird** [comment]
> Do we have documentation for WebSocket? I feel that this PR should not be submitted to the limit conn part

In APISIX, there is currently no special document to describe websocket related functions

**Review by SkyeYoung — APPROVED** [review]

File: docs/zh/latest/plugins/limit-conn.md [hunk 20]
```diff
@@ -136,3 +136,76 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    }
+}'
+```
+
+上述路由在 `/ws` 上开启了 WebSocket 代理，并限制了 WebSocket 连接并发数 1，超过 1 个并发的 WebSocket 连接将返回 `503` 拒绝请求。
```
Comment (guoqqqi) [inline docs/zh/latest/plugins/limit-conn.md:170]:
```suggestion
上述路由在 `/ws` 上开启了 WebSocket 代理，并限制了 WebSocket 连接并发数 1（conn），超过 1 个并发的 WebSocket 连接将返回 `503` 拒绝请求。
```

**Review by bzp2010 — DISMISSED** [review]

File: docs/en/latest/plugins/limit-conn.md [hunk 21]
```diff
@@ -141,3 +141,76 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
     }
 }'
 ```
+
+## Limit the rate of WebSocket connections
```
Comment (moonming) [inline docs/en/latest/plugins/limit-conn.md:145]:
```suggestion
## Limit connections of WebSocket
```

File: docs/zh/latest/plugins/limit-conn.md [hunk 22]
```diff
@@ -136,3 +136,76 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
 ```
 
 现在就已经移除了 limit-conn 插件了。其他插件的开启和移除也是同样的方法。
+
+## 对 WebSocket 连接限速
```
Comment (moonming) [inline docs/zh/latest/plugins/limit-conn.md:140]: `限速` or `限制并发数`？

File: docs/en/latest/plugins/limit-conn.md [hunk 23]
```diff
@@ -141,3 +141,76 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
...
+        "nodes": {
+            "127.0.0.1:1980": 1
+        }
+    }
+}'
+````
+
+The above route enables the WebSocket proxy on `/ws`, and limits the number of concurrent WebSocket connections to 1. More than 1 concurrent WebSocket connection will return `503` to reject the request.
+
+2. Initiate a WebSocket request, and the link is established successfully
```
Comment (moonming) [inline docs/en/latest/plugins/limit-conn.md:177]:
```suggestion
2. Initiate a WebSocket request, and the connection is established successfully
```

File: docs/zh/latest/plugins/limit-conn.md (hunk 20 above)
Comment (soulbird) [reply docs/zh/latest/plugins/limit-conn.md:170]: How about "上述路由在 `/ws` 上开启了 WebSocket 代理，并限制了 WebSocket 连接并发数为 1" ？

File: docs/en/latest/plugins/limit-conn.md (hunk 21 above)
Comment (soulbird) [reply docs/en/latest/plugins/limit-conn.md:145]: How about `Limit the number of concurrent WebSocket connections` ?

**Review by guoqqqi — APPROVED** [review]
LGTTM

**Review by SkyeYoung — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Pull request merged**